### PR TITLE
Clarify wait and wake operators in web embedding

### PR DESCRIPTION
**Pull request by binji**
See issue #25.

### CODE REVIEW

File: proposals/threads/Overview.md [hunk 1]
```diff
@@ -197,9 +197,9 @@ Both wake and wait operators trap if the effective address of either operator
 is misaligned or out-of-bounds. The wait operators requires an alignment of
 their memory access size. The wait operator requires an alignment of 32 bits.
 
-The embedder is also permitted to suspend or wake an agent. A suspended agent
-can be woken by the embedder or the wake operator, regardless of how the agent
-was suspended (e.g. via the embedder or a wait operator).
+For the web embedding, the agent can also be suspended or woken via the
+[`Atomics.wait`][] and [`Atomics.wake`][] functions respectively. An agent 
+will not be suspended or woken for other reasons.
```
Comment (jfbastien) [inline proposals/threads/Overview.md]: I'm not sure that's correct. Just like we're allowed to suspend any main thread of service worker, we can suspend an agent indefinitely, no?

I think you're mixing the "spurious wakeup" discussion here. There's no spurious wakeup from a wait, but we can definitely suspend without a wait being invoked.

File: proposals/threads/Overview.md [hunk 2]
```diff
@@ -243,6 +254,15 @@ returns the number of waiters that were woken as an `i32`.
 | `wake count` > 0 | Wake min(`wake count`, `num waiters`) waiters |
 
   * `wake`: wake up `wake count` threads waiting on the given address via `i32.wait` or `i64.wait`
+  
+For the web embedding, `i32.wait` is equivalent in behavior to executing the following:
```
Comment (jfbastien) [inline proposals/threads/Overview.md]: What about `i64.wait` / `i64.wake`? How do those interact with the web embedding?

Comment (binji) [inline proposals/threads/Overview.md]: There is no `i64.wake`, but yeah. I'll add some info about i64.wait.

File: proposals/threads/Overview.md (hunk 1 above)
Comment (binji) [inline proposals/threads/Overview.md]: Right, as @lars-t-hansen mentions in #25, you can suspend an agent only if you suspend all agents in the cluster. I'll clarify.

File: proposals/threads/Overview.md (hunk 2 above)
Comment (jfbastien) [inline proposals/threads/Overview.md]: Derp right!

File: proposals/threads/Overview.md [hunk 3]
```diff
@@ -228,6 +229,37 @@ another agent wakes this one, this operator returns 2 ("timed-out").
...
+   as described above. The [`Atomics.wait`][] operation is modified:
+   1. `ValidateSharedIntegerTypedArray` will fail if the typed-array type is not an
+      `Int64Array`.
+   1. `value` is not converted to an `Int32`, but kept in a 64-bit integer
+      representation.
+   1. `indexedPosition` is (`i` x 8) + `offset`
+1. Return an `i32` value as described in the above table:
+   ("ok" -> `0`, "not-equal" -> `1`, "timed-out" -> `2`).
 
 ### Wake
```
Comment (jfbastien) [inline proposals/threads/Overview.md]: Do you want to mention that there are no spurious wakes?

Comment (binji) [inline proposals/threads/Overview.md]: Done.

**Comment by binji** [comment]
Seems there are no further concerns. Landing; we can open additional issues if necessary.

**Pull request merged**